src/gtest/example_tests.cpp

```diff
@@ -0,0 +1,66 @@
+#include "generated_wrappers.hpp"
+
+extern "C" {
+    #include "dict.h"
+}
+
+// Use a class name descriptive of your test unit
+class ExampleTest : public ::testing::Test {
+    // standard boilerplate supporting mocked functions
+    protected:
+        MockValkey mock;
+        RealValkey real;
+
+    // The SetUp() function is called before each test.
+    void SetUp() override {
+        memset(&server, 0, sizeof(valkeyServer));
+        server.hz = CONFIG_DEFAULT_HZ;
+    }
+
+    // The TearDown() function is called after each test.
+    void TearDown() override {}
+};
+
+// Include this (should end in "DeathTest") if testing that code asserts/dies.
+using ExampleDeathTest = ExampleTest;
+
+// Example of a DeathTest, which passes only if the code crashes.
+// TEST_F(ExampleDeathTest, TestSimpleDeath) {
+//     EXPECT_DEATH(
+//         {
+//             *(static_cast<volatile char*>(0)) = 'x'; // SEGV
+//         },
+//         ""
+//     );
+// }
+
+// Simple assertions test
+TEST_F(ExampleTest, TestAssertions) {
+    int a = 5, b = 3;
+    const char *str = "hello";
+    // Use EXPECT_ macros to test a condition.  If the value is not as expected, the test will fail.
+    // Use ASSERT_ macros to test a condition AND immediately end the test.
+    // Prefer to use EXPECT_ macros unless the test can't reasonably continue.  This allows multiple 
+    //  conditions to be tested and reported rather than ending at the first unexpected value.
+    EXPECT_EQ(8, a + b);
+    EXPECT_LE(b, a);
+    EXPECT_GT(a, b);
+    EXPECT_STREQ(str, "hello");
+    ASSERT_EQ(2, a - b);
+}
+
+// Test matcher works in custom_matchers.hpp
+TEST_F(ExampleTest, TestMatchers) {
+    robj *robj_str = createStringObject("test", 4);
+    ASSERT_NE(robj_str , nullptr); // "ASSERT" is correct here, because the test can't reasonably continue
+    EXPECT_THAT(robj_str, robjEqualsStr("test"));
+    decrRefCount(robj_str);
+}
+
+// Verify mocking works via zfree
+TEST_F(ExampleTest, TestMocking) {
```

Comment:
I think this is the type of mocking that got ElastiCache into trouble. This is trying to make sure zfree is called, but there are better systemic ways like https://github.com/valkey-io/valkey/blob/0a9cb985c8c37c54c5007bc0af2dcd0b6752a529/src/server.h#L4077.